### PR TITLE
Newmetric: Negative predictive value

### LINKED ISSUE #2415: A new metric NPV
## 🚀 Feature

NPV (Negative Predictive Value) is a metric used to evaluate the accuracy of a medical test or diagnostic method.

### Motivation

NPV (Negative Predictive Value) is a metric used to evaluate the accuracy of a medical test or diagnostic method. It measures the proportion of individuals who are truly negative among those who test negative. In simple terms, NPV represents the percentage of true negatives in the population tested negative.

The formula to calculate NPV is as follows:

NPV = TN / (TN + FN)

Here, True Negatives refer to the number of individuals who are actually negative and test negative, while False Negatives represent the number of individuals who are actually negative but test positive (false negative errors).

### PR DESCRIPTION
**Pull request by SkafteNicki**
## What does this PR do?

Fixes #2415 

<details>
  <summary>Before submitting</summary>

- [ ] Was this **discussed/agreed** via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/torchmetrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to **update the docs**?
- [ ] Did you write any new **necessary tests**?

</details>

<details>
  <summary>PR review</summary>

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

</details>

## Did you have fun?

Make sure you had fun coding 🙃

----
📚 Documentation preview 📚: https://torchmetrics--2433.org.readthedocs.build/en/2433/

### CODE REVIEW

File: src/torchmetrics/classification/negative_predictive_value.py [hunk 1]
```diff
@@ -0,0 +1,516 @@
...
+
+    """
+
+    plot_lower_bound: float = 0.0
+    plot_upper_bound: float = 1.0
+
+    def compute(self) -> Tensor:
+        """Compute metric."""
+        tp, fp, tn, fn = self._final_state()
+        return _specificity_reduce(tp, fp, tn, fn, average="binary", multidim_average=self.multidim_average)
```
Comment (codyhesse) [inline src/torchmetrics/classification/negative_predictive_value.py:108]: Shouldn't this be _negative_predictive_value_reduce?

Comment (Asunatan) [reply src/torchmetrics/classification/negative_predictive_value.py:108]: bug
There's something wrong here.
You may need to redefine a new function: _negative_predictive_value_reduce.
NPV = TN / (TN + FN)

Comment (codyhesse) [reply src/torchmetrics/classification/negative_predictive_value.py:108]: _negative_predictive_value_reduce has been defined [here](https://github.com/Lightning-AI/torchmetrics/blob/32e35d1d0450d01b6269e09a4e1611973851d250/src/torchmetrics/functional/classification/negative_predictive_value.py#L37), but isn't used on this line for some reason

File: tests/unittests/classification/test_negative_predictive_value.py [hunk 2]
```diff
@@ -0,0 +1,600 @@
...
+
+from unittests import NUM_CLASSES, THRESHOLD
+from unittests.classification._inputs import _binary_cases, _multiclass_cases, _multilabel_cases
+from unittests.helpers import seed_all
+from unittests.helpers.testers import MetricTester, inject_ignore_index
+
+seed_all(42)
+
+
+def _calc_negative_predictive_value(tn, fp):
```
Comment (Asunatan) [inline tests/unittests/classification/test_negative_predictive_value.py]: fn?

**Review by Asunatan — COMMENTED** [review]
Thank you for your team's significant contribution to calculating negative predictive value (NPV) as a new classification metric. This work can reduce the computational burden on clinical doctors.

**Review by Borda — APPROVED** [review]

File: src/torchmetrics/classification/negative_predictive_value.py [hunk 3]
```diff
@@ -0,0 +1,521 @@
...
+            tp, fp, tn, fn, average=self.average, multidim_average=self.multidim_average, multilabel=True
+        )
+
+    def plot(
+        self, val: Optional[Union[Tensor, Sequence[Tensor]]] = None, ax: Optional[_AX_TYPE] = None
+    ) -> _PLOT_OUT_TYPE:
+        """Plot a single or multiple values from the metric.
+
+        Args:
+            val: Either a single result from calling `metric.forward` or `metric.compute` or a list of these results.
```
Comment (Borda) [inline src/torchmetrics/classification/negative_predictive_value.py:421]:
```suggestion
            val: Either a single result from calling ``metric.forward`` or ``metric.compute`` or a list of these results.
```

**Review by Borda — APPROVED** [review]

File: src/torchmetrics/functional/classification/negative_predictive_value.py [hunk 4]
```diff
@@ -0,0 +1,419 @@
...
+    Args:
+        preds: Tensor with predictions
+        target: Tensor with true labels
+        num_classes: Integer specifying the number of classes
+        average:
+            Defines the reduction that is applied over labels. Should be one of the following:
+
+            - ``micro``: Sum statistics over all labels
+            - ``macro``: Calculate statistics for each label and average them
+            - ``weighted``: calculates statistics for each label and computes weighted average using their support
```
Comment (lantiga) [inline src/torchmetrics/functional/classification/negative_predictive_value.py:173]:
```suggestion
            - ``weighted``: Calculate statistics for each label and compute a weighted average using their support
```

File: src/torchmetrics/functional/classification/negative_predictive_value.py [hunk 5]
```diff
@@ -0,0 +1,419 @@
...
+            - ``"none"`` or ``None``: calculates statistic for each label and applies no reduction
```
Comment (lantiga) [inline src/torchmetrics/functional/classification/negative_predictive_value.py:174]:
```suggestion
            - ``"none"`` or ``None``: Calculate statistics for each label and apply no reduction
```

File: src/torchmetrics/functional/classification/negative_predictive_value.py [hunk 6]
```diff
@@ -0,0 +1,419 @@
...
+        multidim_average:
+            Defines how additionally dimensions ``...`` should be handled. Should be one of the following:
+
+            - ``global``: Additional dimensions are flatted along the batch dimension
+            - ``samplewise``: Statistic will be calculated independently for each sample on the ``N`` axis.
+              The statistics in this case are calculated over the additional dimensions.
+
+        ignore_index:
+            Specifies a target value that is ignored and does not contribute to the metric calculation
+        validate_args: bool indicating if input arguments and tensors should be validated for correctness.
```
Comment (lantiga) [inline src/torchmetrics/functional/classification/negative_predictive_value.py:188]:
```suggestion
        validate_args: Bool indicating if input arguments and tensors should be validated for correctness.
```

File: src/torchmetrics/functional/classification/negative_predictive_value.py [hunk 7]
```diff
@@ -0,0 +1,419 @@
...
+        preds: Tensor with predictions
+        target: Tensor with true labels
+        num_labels: Integer specifying the number of labels
+        threshold: Threshold for transforming probability to binary (0,1) predictions
+        average:
+            Defines the reduction that is applied over labels. Should be one of the following:
+
+            - ``micro``: Sum statistics over all labels
+            - ``macro``: Calculate statistics for each label and average them
+            - ``weighted``: calculates statistics for each label and computes weighted average using their support
```
Comment (lantiga) [inline src/torchmetrics/functional/classification/negative_predictive_value.py:288]:
```suggestion
            - ``weighted``: Calculate statistics for each label and compute a weighted average using their support
```

File: src/torchmetrics/functional/classification/negative_predictive_value.py [hunk 8]
```diff
@@ -0,0 +1,419 @@
...
+            - ``"none"`` or ``None``: calculates statistic for each label and applies no reduction
```
Comment (lantiga) [inline src/torchmetrics/functional/classification/negative_predictive_value.py:289]:
```suggestion
            - ``"none"`` or ``None``: Calculate statistics for each label and apply no reduction
```

File: src/torchmetrics/functional/classification/negative_predictive_value.py [hunk 9]
```diff
@@ -0,0 +1,419 @@
...
+        multidim_average:
+            Defines how additionally dimensions ``...`` should be handled. Should be one of the following:
+
+            - ``global``: Additional dimensions are flatted along the batch dimension
+            - ``samplewise``: Statistic will be calculated independently for each sample on the ``N`` axis.
+              The statistics in this case are calculated over the additional dimensions.
+
+        ignore_index:
+            Specifies a target value that is ignored and does not contribute to the metric calculation
+        validate_args: bool indicating if input arguments and tensors should be validated for correctness.
```
Comment (lantiga) [inline src/torchmetrics/functional/classification/negative_predictive_value.py:300]:
```suggestion
        validate_args: Bool indicating if input arguments and tensors should be validated for correctness.
```

**Review by lantiga — APPROVED** [review]
LGTM, just a few nits on the docstrings

**Pull request merged**